python_modules/dagster/dagster/_config/config_type.py

```diff
@@ -404,7 +404,9 @@ def __init__(
     ):
         from .field import resolve_to_config_type
 
-        self.scalar_type = resolve_to_config_type(scalar_type)
+        self.scalar_type = check.inst(
+            cast(ConfigType, resolve_to_config_type(scalar_type)), ConfigType
```

Comment:
FWIW this cast here certainly convinced me that the "type verifying" approach is boilerplate-y and pedantic.